xarray/core/dataarray.py

```diff
@@ -440,7 +440,7 @@ def subset(dim, label):
         variables = {label: subset(dim, label) for label in self.get_index(dim)}
         variables.update({k: v for k, v in self._coords.items() if k != dim})
         indexes = propagate_indexes(self._indexes, exclude=dim)
-        coord_names = set(self._coords) - set([dim])
+        coord_names = set(self._coords) - {dim}
```

Comment:
Could this be a problem if dim is a string?

Reply:
works on 3.8: ![image](https://user-images.githubusercontent.com/2448579/90990304-0333d000-e590-11ea-8c86-4fc44b4f5d31.png)